### PR TITLE
Change prefix for _importlib_modulespec names

### PR DESCRIPTION
**Pull request by pkch**
(Not ready for merging)

This PR is meaningless without python/mypy#3107; if that one is approved, this would fix the type reported for modules from `_importlib_modulespec.ModuleType` to `types.ModuleType`. (There is no dependency in the other direction.)

This PR is related to python/mypy#3106.

### CODE REVIEW

**Comment by pkch** [comment]
Next commit would break mypy unless python/mypy#3107 is merged.

Note: I wasn't sure about this line in 2/abc.pyi:

    types = ...  # type: module

which I changed to

    types = ...  # type: ModuleType

I'm afraid I might be breaking something even though the tests pass.

Also, werkzeug stubs had:

    class module(ModuleType):
      def __getattr__(self, name): ...
      def __dir__(self): ...

I removed them, but I also suspect they may serve some non-obvious purpose.

File: stdlib/2/tokenize.pyi [hunk 1]
```diff
@@ -102,15 +103,15 @@ chain = ...  # type: type
 double3prog = ...  # type: type
 endprogs = ...  # type: Dict[str, Any]
 pseudoprog = ...  # type: type
-re = ...  # type: module
+re = ...  # type: ModuleType
```
Comment (JelleZijlstra) [inline stdlib/2/tokenize.pyi]: This probably shouldn't be here at all. (I'm guessing tokenize just imports re.)

File: stdlib/3/_importlib_modulespec.pyi [hunk 2]
```diff
@@ -9,6 +9,8 @@ from abc import ABCMeta
 import sys
 from typing import Any, Optional
 
+MYPY_MODULE = 'importlib.machinery'
```
Comment (JelleZijlstra) [inline stdlib/3/_importlib_modulespec.pyi]: If we go down this route, we should prefix it with `_` to make clear that this name isn't actually part of this module. However, I'd prefer to just hardcode the location of ModuleType in mypy and not add special handling in typeshed.

**Comment by pkch** [comment]
I don't have an opinion on this, I would be happy to hard code it in mypy. Just wasn't sure if it's ok given that it's hard to find later when more circular import definitions happen.

File: stdlib/3/_importlib_modulespec.pyi (hunk 2 above)
Comment (pkch) [inline stdlib/3/_importlib_modulespec.pyi]: I don't have an opinion on this, I would be happy to hard code it in mypy. Just wasn't sure if it's ok given that it's hard to find later when more circular import definitions happen.

**Comment by pkch** [comment]
I replaced the MYPY_MODULE with hard-coded remapping; so this PR simply replaces `module` with `types.ModuleType`. It is blocked by python/mypy#3107

**Comment by pkch** [comment]
Superceded by #1156

**Pull request closed without merging**